src/impl/collection_change_builder.cpp

```diff
@@ -142,7 +142,12 @@ void CollectionChangeBuilder::parse_complete()
     for (auto move : m_move_mapping) {
         REALM_ASSERT_DEBUG(deletions.contains(move.second));
         REALM_ASSERT_DEBUG(insertions.contains(move.first));
-        moves.push_back({move.second, move.first});
+        if (move.first == move.second) {
+            deletions.remove(move.second);
+            insertions.remove(move.first);
+        }
+        else
```

Comment:
Uh... it is merged from #158 to solve the build problems with 2.0.0-rc4 ... We are not using those code core :) @kneth